### PR TITLE
CBP-1301 gin app

### PR DESCRIPTION
**Pull request by JesusRamirez2910**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,56 @@
...
+
+`sdkKey := "<INSERT YOUR SDK KEY HERE>"`
+
+For example:
+
+`sdkKey := "8993020e-78cd-4ea0-7cfe-e1a64112eddb"`
+
+## Run the application
+
+Using your favourite package manager:
```
Comment (markawm) [inline README.md:20]: Copy & paste? Not really much choice of package managers for Go :-)

File: main.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+func main() {
+	router := gin.Default()
+	initFlags()
+
+	router.GET("/", func(c *gin.Context) {
+		c.JSON(http.StatusOK, gin.H{"message": "Hello, You Created a Web App!"})
+	})
+
+	router.GET("/demo", func(c *gin.Context) {
+		ctx := context.NewContext(map[string]interface{}{"email": "[email protected]"})
```
Comment (markawm) [inline main.go:34]: I like that this shows an example of using a context for flag evaluation. But it seems to stop _just_ short of actually showing it in action.
e.g. if we added an `email` parameter to the request then you could use _that_ dynamic value in the context and have the flag show different values for different users.
How about having the simple GET endpoint like this evaluating a flag without a context (`IsEnabled(nil)`), and another POST endpoint that has a/some query/body parameter(s) that you then use in the flag context?

**Review by markawm — COMMENTED** [review]
Just another thought: could you show an example of [clean shutdown](https://docs.cloudbees.com/docs/cloudbees-feature-management-api/latest/api-reference/go-api#_shutdown) here? Not all SDKs have this, but since the Go one does...

**Comment by JesusRamirez2910** [comment]
@markawm, updated the PRs, please lmk if any other change is needed.

**Review by markawm — APPROVED** [review]
All lgtm for this change. 🎉 
Are you going to follow up with the other suggestions (evaluate with context, shutdown)?

**Comment by JesusRamirez2910** [comment]
> All lgtm for this change. 🎉 Are you going to follow up with the other suggestions (evaluate with context, shutdown)?

yes, I'll look into that. Thanks!

**Pull request merged**